dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdFileUploadTask.kt

```diff
@@ -0,0 +1,314 @@
+                sourceSetRoots,
+                remoteRepositoryUrl
+            )
+        }
+        if (repositories.isNotEmpty()) {
+            generateRepositoryFile(repositories)
+        }
+
+        val site = DatadogSite.valueOf(site)
+        for (mappingFile in mappingFiles) {
+            LOGGER.info("Uploading ${mappingFile.fileType} file: ${mappingFile.file.absolutePath}")
+            uploader.upload(
+                site,
+                mappingFile,
+                if (repositories.isEmpty()) null else repositoryFile,
+                apiKey,
+                DdAppIdentifier(
+                    serviceName = serviceName,
+                    version = versionName,
+                    versionCode = versionCode,
+                    variant = variantName,
+                    buildId = buildId.get()
+                ),
+                repositories.firstOrNull(),
+                !disableGzipOption.isPresent
+            )
+        }
+    }
+
+    internal fun configureWith(
+        apiKey: ApiKey,
+        extensionConfiguration: DdExtensionConfiguration,
+        variant: ApplicationVariant
+    ) {
+        this.apiKey = apiKey.value
+        apiKeySource = apiKey.source
+        site = extensionConfiguration.site ?: ""
+
+        versionName = variant.versionName ?: ""
+        versionCode = variant.versionCode
+        serviceName = extensionConfiguration.serviceName ?: variant.applicationId
+        variantName = variant.flavorName
+        remoteRepositoryUrl = extensionConfiguration.remoteRepositoryUrl ?: ""
+    }
+
+    private fun applySiteFromEnvironment() {
+        val environmentSite = System.getenv(DATADOG_SITE)
+        if (!environmentSite.isNullOrEmpty()) {
+            if (this.site.isNotEmpty()) {
+                DdAndroidGradlePlugin.LOGGER.info(
+                    "Site property found as DATADOG_SITE env variable, but it will be ignored," +
+                        " because also an explicit one was provided in extension."
+                )
+                return
+            }
+            val site = DatadogSite.fromDomain(environmentSite)
+            if (site == null) {
+                DdAndroidGradlePlugin.LOGGER.warn("Unknown Datadog domain provided: $environmentSite, ignoring it.")
+            } else {
+                DdAndroidGradlePlugin.LOGGER.info("Site property found in Datadog CI config file, using it.")
+                this.site = site.name
+            }
+        }
+    }
+
+    private fun applyDatadogCiConfig(datadogCiFile: File) {
+        try {
+            val config = JSONObject(datadogCiFile.readText())
+            applyApiKeyFromDatadogCiConfig(config)
+            applySiteFromDatadogCiConfig(config)
+        } catch (e: JSONException) {
+            DdAndroidGradlePlugin.LOGGER.error("Failed to parse Datadog CI config file.", e)
+        }
+    }
+
+    private fun applyApiKeyFromDatadogCiConfig(config: JSONObject) {
```

Comment:
I think to be cleaner we can apply the same logic as for the site below to maintain a pattern.

Reply:
Probably. This was straight copied from DdMappingFileUploadTask, but maybe I can revisit here and make it cleaner.